packages/ckeditor5-dev-tests/lib/utils/manual-tests/compilehtmlfiles.js

```diff
@@ -80,8 +92,41 @@ function uniq( arr ) {
 	return Array.from( new Set( arr ) );
 }
 
-function changeExtension( file, newExt ) {
+function setExtension( file, newExt ) {
 	const { dir, name } = path.parse( file );
 
 	return path.join( dir, name + '.' + newExt );
 }
+
+function getFilePathWithoutExtension( file ) {
+	const { dir, name } = path.parse( file );
+
+	return path.join( dir, name );
+}
+
+function watchFiles( filePaths, onChange ) {
+	for ( const filePath of filePaths ) {
+		watchFile( filePath, onChange );
+	}
+}
+
+function watchFile( filePath, onChange ) {
+	const debouncedOnChange = debounce( () => onChange( filePath ), 500 );
+	fs.watch( filePath, debouncedOnChange );
+}
+
+function debounce( callback, delay ) {
```

Comment:
Don't implement such methods yourself.

Either use a watcher which supports this or use lodash's deboune.